src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,134 @@
+
     @Override
     public String getString() {
-        return null;
+        return string;
     }
 
     @Override
     public void setString(String string) {
+        this.string = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        if (string == null)
+            return 0;
+        if (string.length() == 0)
+            return 0;
+
+        int count = 0;
+        for (int i = 0; i < string.length() - 2; i++) {
+            if (i == 0 && !isSpaceChar(string.charAt(i)))
+                count++;
+            else if (isSpaceChar(string.charAt(i)) && !isSpaceChar(string.charAt(i + 1)))
+                count++;
+        }
+
+        return count;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        if (n <= 0)
+            throw new IllegalArgumentException();
+        if (n > string.length())
+            throw new IndexOutOfBoundsException();
+
+        StringBuilder newStr = new StringBuilder();
+        for (int i = 1; i < string.length() + 1; i++) {
+            if (i % n == 0) {
+                if (maintainSpacing)
+                    newStr.append(' ');
+            } else
+                newStr.append(string.charAt(i - 1));
+        }
+
+        return newStr.toString();
     }
 
     @Override
     public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+        if (count() < endWord)
+            throw new IndexOutOfBoundsException();
+        if (startWord <= 0 || endWord <= 0 || startWord > endWord)
+            throw new IllegalArgumentException();
+
+        String[] words = new String[(endWord - startWord) + 1];
+        int wordIndex = 0;
+
+        StringBuilder sb = new StringBuilder();
+        int wordCount = 0;
+        for (int i = 0; i < string.length() - 1; i++) {
+            // Case where first character is a word
+            if (i == 0 && !isSpaceChar(string.charAt(i))) {
+                wordCount++;
+                if (wordCount == startWord)
+                    sb.append(string.charAt(i));
+
+                // Case where beginning of new word is found
+            } else if (isSpaceChar(string.charAt(i)) && !isSpaceChar(string.charAt(i + 1))) {
+                wordCount++;
+                i++;
+                if (wordCount >= startWord && wordCount <= endWord)
+                    sb.append(string.charAt(i));
+
+                // Case where end of word is found
+            } else if (!isSpaceChar(string.charAt(i)) && isSpaceChar(string.charAt(i + 1))) {
+                if (wordCount >= startWord && wordCount <= endWord) {
+                    sb.append(string.charAt(i));
+                    words[wordIndex++] = sb.toString();
+                    sb = new StringBuilder();
+                }
+
+            } else if (!isSpaceChar(string.charAt(i)) && wordCount >= startWord && wordCount <= endWord) {
+                sb.append(string.charAt(i));
```

Comment:
StringBuilder is a good choice, I however favored the split method of the string class  using the regex //s+, this gives you an array of all the char groups between whitespace gaps, then you can just loop through that array and add desired words to your final words array. that solution is a bit shorter to write and simpler IMHO